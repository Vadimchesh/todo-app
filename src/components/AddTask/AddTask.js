import React from 'react';
import { useDispatch } from 'react-redux';
import styles from './style.module.scss';

const AddTask = () => {
  const dispatch = useDispatch();
  const handleKeyDown = (e) => {
    const trimmedText = e.target.value.trim();
    if (e.which === 13 && trimmedText) {
      dispatch({ type: 'todoAdd', payload: trimmedText });
      e.target.value = '';
    }
  };

  return (
    <div className={styles.container}>
      <span className={styles.arrow} />
      <input
        autoFocus
        className='new-todo'
        placeholder='What needs to be done?'
        onKeyDown={handleKeyDown}
      />
    </div>
  );
};

export default AddTask;
