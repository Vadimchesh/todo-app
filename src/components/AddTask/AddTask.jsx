import React from "react";
import styles from "./style.module.scss";

function AddTask() {
  return (
    <div className={styles.container}>
      <span className={styles.arrow} />
      <input type="text" placeholder="Whats needs to be done?" />
    </div>
  );
}

export default AddTask;
