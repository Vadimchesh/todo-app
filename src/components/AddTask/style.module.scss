@import "../../assets/scss/variables.scss";

.container {
  @include container;
  position: relative;
  box-shadow: inset 0px -2px 8px -4px rgba(0, 0, 0, 0.13);
}
.container input {
  font-size: $main-size;
  font-style: italic;
  font-family: $main-font;
  color: $gray-color;
  width: 100%;
  border: 0px;
  padding-left: 2.5rem;
  outline: none;
}

.arrow {
  position: absolute;
  top: 30%;
  width: 1.2rem;
  height: 1.2rem;
  outline: none;
  border: none;
  background-color: transparent;
  box-shadow: -3px 3px 0 rgba(0, 0, 0, 0.5);
  transform: rotate(-45deg);
}
