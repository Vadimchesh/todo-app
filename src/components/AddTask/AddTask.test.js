import { shallow, ShallowWrapper } from 'enzyme';
import React from 'react';
import { useDispatch } from 'react-redux';
import AddTask from './AddTask';

const mockDispatch = jest.fn();

jest.mock('react-redux', () => {
  const originalModule = jest.requireActual('react-redux');
  return {
    ...originalModule,
    useDispatch: jest.fn(() => mockDispatch),
  };
});

describe('Add Task:', () => {
  let wrapper = ShallowWrapper;
  beforeEach(() => {
    wrapper = shallow(<AddTask />);
  });
  afterAll(() => {
    jest.resetAllMocks();
  });
  it('snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('called handleKeyDown fn', () => {
    wrapper
      .find('input')
      .simulate('keydown', { which: 13, target: { value: 'test' } });
    expect(useDispatch).toBeCalledTimes(1);
    expect(mockDispatch).toBeCalledWith({
      type: 'todoAdd',
      payload: 'test',
    });
  });
});
