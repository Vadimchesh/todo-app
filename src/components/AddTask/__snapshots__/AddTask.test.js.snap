// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add Task: snapshot 1`] = `
<div
  className="container"
>
  <span
    className="arrow"
  />
  <input
    autoFocus={true}
    className="new-todo"
    data-cy="inputText"
    onKeyDown={[Function]}
    placeholder="What needs to be done?"
  />
</div>
`;
